### PR TITLE
Functional Tests -  Enable/Disable tax breakdown

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add Functional tests for Enable/Disable tax breakdown in the invoice
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/orders/invoices/03_invoiceOptions/02_enableDisableTaxBreakdown.js" URL_FO=yourShopURL DOWNLOAD_PATH=yourDownloadPath npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16747)

### CODE REVIEW

File: tests/puppeteer/pages/BO/orders/view.js [hunk 1]
```diff
@@ -62,4 +65,25 @@ module.exports = class Order extends BOBasePage {
...
+  }
+
+  /**
+   * Download invoice
+   * @returns {Promise<void>}
+   */
+  async downloadInvoice() {
+    /* eslint-disable */
+    await this.page.$eval(this.documentNumberLink, el => el.target = '');
+    await this.page.click(this.documentNumberLink);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/orders/view.js]: Can you a comment here please, why target should be deleted before click.

File: tests/puppeteer/pages/BO/international/taxes/taxRules/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require('module-alias/register');
+const BOBasePage = require('@pages/BO/BObasePage');
+
+module.exports = class TaxRules extends BOBasePage {
+  constructor(page) {
+    super(page);
+
+    this.pageTitle = 'Tax Rules • prestashop';
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/international/taxes/taxRules/index.js]:
```suggestion
    this.pageTitle = 'Tax Rules • ';
```
The `prestashop` part is the name of the shop :)

**Review by SimonGrn — COMMENTED** [review]
Thanks @nesrineabdmouleh

File: tests/puppeteer/pages/FO/searchResults.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    this.searchResultProductArticle = `${this.searchResultSection} div:nth-child(%NUMBER) > article`;
+    this.productImg = `${this.searchResultProductArticle} img`;
+  }
+
+  /**
+   * Go to the product page
+   * @param id, product id
+   */
+  async goToProductPage(id) {
+    await this.waitForSelectorAndClick(this.productImg.replace('%NUMBER', id));
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/searchResults.js]: Maybe you should wait for navigation here :thinking: 
```suggestion
    await Promise.all([
      this.waitForSelectorAndClick(this.productImg.replace('%NUMBER', id)),
      this.page.waitForNavigation({waitUntil:'networkidle0'}),
    ]);
```

File: tests/puppeteer/pages/BO/international/taxes/taxRules/add.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+require('module-alias/register');
+const BOBasePage = require('@pages/BO/BObasePage');
+
+module.exports = class AddTaxRules extends BOBasePage {
+  constructor(page) {
+    super(page);
+
+    this.pageTitleCreate = 'Tax Rules > Add new • prestashop';
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/international/taxes/taxRules/add.js]:
```suggestion
    this.pageTitleCreate = 'Tax Rules > Add new •';
```
depends of installation values

File: tests/puppeteer/campaigns/utils/files.js [hunk 5]
```diff
@@ -25,4 +26,34 @@ module.exports = {
...
     return found;
   },
+
+  /**
+   * Get page text
+   * @param pdf
+   * @param pageNo
+   * @return text, text in PDF file
+   */
+  async getPageText(pdf, pageNo) {
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/utils/files.js]:
```suggestion
  async getPageTextFromPdf(pdf, pageNo) {
```

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**